spec/ruby/core/string/encode_spec.rb

```diff
@@ -19,13 +19,17 @@
     it "returns a copy when Encoding.default_internal is nil" do
       Encoding.default_internal = nil
       str = "あ"
-      str.encode.should_not equal(str)
+      encoded = str.encode
+      encoded.should_not equal(str)
+      encoded.should eql(str)
```

Comment:
This should probably be `encoded.should == str`, unless there is a specific reason to use Hash equality (`#eql?`) here?

Reply:
You're right; could you please help fix the 7 occurrences of eql? Thank you! I thought eql and == were equivalent for arrays and arbitrarily picked one.

Reply:
@LillianZ Yes, I'll update these.

Reply:
`eql?` are updated  to `==` in the spec here: https://github.com/oracle/truffleruby/commit/68825d1df0eaa0e7355b8e027f03ab066cecda71